iOS release flow automation

kshann opened the pull request:
<!--
Note: This checklist is a reminder of our shared engineering expectations. Feel free to change it, although assigning a GitHub reviewer and the items in bold are required.

⚠️ If you're an external contributor, please file an issue first before working on a PR, as we can't guarantee that we will accept your changes if they haven't been discussed ahead of time. Thanks!
-->

Task/Issue URL: https://app.asana.com/0/1205419239275955/1208056445209269/f
Tech Design URL:
CC:

**Description**:
Initial automation sharing for iOS release flow based on macOS release flow. 

- [x] Set schedule for release bump
- [x] Bump fastlane plugin
- [x] Align build number before merging. 

<!--
If at any point it isn't actively being worked on/ready for review/otherwise moving forward strongly consider closing it (or not opening it in the first place). If you decide not to close it, use Draft PR while work is still in progress or use `DO NOT MERGE` label to clarify the PRs state and comment with more information.
-->

**Steps to test this PR**:
1. Run a release flow
2.  [Alpha build to test version and buildnumber config](https://github.com/duckduckgo/iOS/actions/runs/12824908763)

<!--
Before submitting a PR, please ensure you have tested the combinations you expect the reviewer to test, then delete configurations you *know* do not need explicit testing.

Using a simulator where a physical device is unavailable is acceptable.
-->

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

kshann committed:
initial release pre requisites

kshann committed:
update configuration

kshann committed:
allow a good build number i guess

kshann committed:
allow a good build number i guess

kshann committed:
dynamic root plist

kshann committed:
make config mandatory

kshann committed:
make config mandatory

kshann committed:
bump version

kshann committed:
Merge branch 'kieran/test-dynamic-root-plist' into kieran/ios-release-flow-automation

kshann committed:
New iOS automated workflows

kshann committed:
bump build number

kshann committed:
align version and build number

kshann committed:
update gha job name

ayoy commented on .github/workflows/ios-code-freeze.yml:
```diff
@@ -0,0 +1,99 @@
+      MM_WEBHOOK_URL: ${{ secrets.MM_WEBHOOK_URL }}
+      SSH_PRIVATE_KEY_FASTLANE_MATCH: ${{ secrets.SSH_PRIVATE_KEY_FASTLANE_MATCH }}
+
+  tag_and_merge:
+    name: Tag and Merge Branch
+    needs: [ create_release_branch, prepare_release ]
+    uses: ./.github/workflows/ios-tag-release.yml
+    with:
+      asana-task-url: ${{ needs.create_release_branch.outputs.asana_task_url }}
+      base-branch: ${{ github.ref_name }}
+      branch: ${{ needs.create_release_branch.outputs.release_branch_name }}
+      prerelease: true
+    secrets:
+      ASANA_ACCESS_TOKEN: ${{ secrets.ASANA_ACCESS_TOKEN }}
+      GHA_ELEVATED_PERMISSIONS_TOKEN: ${{ secrets.GHA_ELEVATED_PERMISSIONS_TOKEN }}
```
The one thing that Publish DMG Release workflow on macOS does for all releases is posting a comment to Asana saying that a new build is available for internal testing (listing tasks added since last internal release). Now I'm thinking that we should ideally call `ios-tag-release-update-asana` here (and restore "internal" `release-type`), instead of `ios-tag-release`.

ayoy reviewed (commented):
LGTM with the exception of "update asana" part that should be happening for every internal bump.

ayoy commented:
@kshann I've opened https://github.com/duckduckgo/iOS/pull/3819 towards your branch. It turned out that we were also missing one script. Happy to connect Monday morning EU time to go through the changes. Thanks!

ayoy committed:
Update Asana after every internal release (#3819)

<!--
Note: This checklist is a reminder of our shared engineering
expectations. Feel free to change it, although assigning a GitHub
reviewer and the items in bold are required.

⚠️ If you're an external contributor, please file an issue first before
working on a PR, as we can't guarantee that we will accept your changes
if they haven't been discussed ahead of time. Thanks!
-->

Task/Issue URL:
https://app.asana.com/0/1205419239275955/1208056445209269/f

**Description**:
Update Code Freeze and Bump Internal Release workflows to call
ios-tag-release-update-asana.yml.
Copy update_asana_for_release.sh over from macOS repo and strip it only
leaving out the functionality
of listing tasks since last internal release - this is used by
ios-tag-release-update-asana.yml

<!--
If at any point it isn't actively being worked on/ready for
review/otherwise moving forward strongly consider closing it (or not
opening it in the first place). If you decide not to close it, use Draft
PR while work is still in progress or use `DO NOT MERGE` label to
clarify the PRs state and comment with more information.
-->

<!--
Before submitting a PR, please ensure you have tested the combinations
you expect the reviewer to test, then delete configurations you *know*
do not need explicit testing.

Using a simulator where a physical device is unavailable is acceptable.
-->

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of
Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering
Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design
Template](https://app.asana.com/0/59792373528535/184709971311943)

kshann committed:
small fixes and pr feedback

kshann committed:
schedule for internal bump

kshann committed:
remove unecessary fetch depth

ayoy reviewed (approved):
Looks great :) thanks @kshann!
